edge-apps/google-calendar/src/events.ts

```diff
@@ -0,0 +1,100 @@
+import { VIEW_MODE } from '@/constants'
+import type { CalendarEvent, ViewMode } from '@/constants'
+import { useSettingsStore } from '@/stores/settings'
+import dayjs from 'dayjs'
+import utc from 'dayjs/plugin/utc'
+import dayJsTimezone from 'dayjs/plugin/timezone'
+
+dayjs.extend(utc)
+dayjs.extend(dayJsTimezone)
+
+const getDateRangeForViewMode = (viewMode: ViewMode) => {
+  const settingsStore = useSettingsStore()
+  const timezone = settingsStore.overrideTimezone || 'UTC'
+
+  // Use dayjs to get current time in the target timezone, ignoring browser timezone
+  const nowInTimezone = dayjs().tz(timezone)
+  const todayInTimezone = nowInTimezone.startOf('day')
+
+  let startDate: Date
+  let endDate: Date
+
+  if (viewMode === VIEW_MODE.DAILY) {
+    // For daily view, start at midnight in the target timezone
+    startDate = todayInTimezone.toDate()
+    endDate = todayInTimezone.add(1, 'day').toDate()
+  } else if (viewMode === VIEW_MODE.WEEKLY) {
+    // For weekly view, show full week starting from Sunday in the target timezone
+    const weekStart = todayInTimezone.startOf('week')
+    startDate = weekStart.toDate()
+    endDate = weekStart.add(7, 'days').toDate()
+  } else if (viewMode === VIEW_MODE.MONTHLY) {
+    // For monthly view, show full month starting from the first day of the month
+    const monthStart = todayInTimezone.startOf('month')
+    startDate = monthStart.toDate()
+    endDate = monthStart.add(1, 'month').toDate()
+  } else {
+    // Default to daily view
+    startDate = todayInTimezone.toDate()
+    endDate = todayInTimezone.add(1, 'day').toDate()
+  }
+
+  return { startDate, endDate }
+}
+
+export const fetchCalendarEventsFromGoogleAPI = async (
+  accessToken: string,
+): Promise<CalendarEvent[]> => {
+  const { calendar_mode: viewMode } = screenly.settings
+  const { startDate, endDate } = getDateRangeForViewMode(viewMode as ViewMode)
+
+  // Fetch events from Google Calendar API
+  const settingsStore = useSettingsStore()
+  const calendarId = settingsStore.calendarId
+  const encodedCalendarId = encodeURIComponent(calendarId as string)
+  const timeMin = startDate.toISOString()
+  const timeMax = endDate.toISOString()
+  const apiUrl = `https://www.googleapis.com/calendar/v3/calendars/${encodedCalendarId}/events?timeMin=${encodeURIComponent(timeMin)}&timeMax=${encodeURIComponent(timeMax)}&singleEvents=true&orderBy=startTime`
+
+  let response
+  let data
+
+  try {
+    response = await fetch(apiUrl, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    })
+
+    if (!response.ok) {
+      throw new Error(
+        'Failed to fetch calendar events from Google Calendar API',
+      )
+    }
+
+    data = await response.json()
+  } catch {
+    return []
+  }
```

Comment:
What are we catching here?

If there is an error, I don't want to render the calendar, @nicomiguelino 

Let it propagate